address build warnings in ncurses example

just a self-reminder.
```
/home/runner/work/RF24Gateway/RF24Gateway/RF24Gateway/examples/ncurses/RF24Gateway_ncurses.cpp: In function ‘void drawTopology()’:
/home/runner/work/RF24Gateway/RF24Gateway/RF24Gateway/examples/ncurses/RF24Gateway_ncurses.cpp:375:25: warning: variable ‘y’ set but not used [-Wunused-but-set-variable]
  375 |                     int y = 0;
      |                         ^
/home/runner/work/RF24Gateway/RF24Gateway/RF24Gateway/examples/ncurses/RF24Gateway_ncurses.cpp:397:25: warning: variable ‘y’ set but not used [-Wunused-but-set-variable]
  397 |                     int y = 0;
      |                         ^
/home/runner/work/RF24Gateway/RF24Gateway/RF24Gateway/examples/ncurses/RF24Gateway_ncurses.cpp:419:25: warning: variable ‘y’ set but not used [-Wunused-but-set-variable]
  419 |                     int y = 0;
      |                         ^
/home/runner/work/RF24Gateway/RF24Gateway/RF24Gateway/examples/ncurses/RF24Gateway_ncurses.cpp: In function ‘void drawRF24Pad()’:
/home/runner/work/RF24Gateway/RF24Gateway/RF24Gateway/examples/ncurses/RF24Gateway_ncurses.cpp:667:42: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 3 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=]
  667 |     wprintw(rf24Pad, "RX Packets(user): %u\n", networkPacketsRX);
      |                                         ~^     ~~~~~~~~~~~~~~~~
      |                                          |     |
      |                                          |     size_t {aka long unsigned int}
      |                                          unsigned int
      |                                         %lu
/home/runner/work/RF24Gateway/RF24Gateway/RF24Gateway/examples/ncurses/RF24Gateway_ncurses.cpp: In function ‘void drawConnPad()’:
/home/runner/work/RF24Gateway/RF24Gateway/RF24Gateway/examples/ncurses/RF24Gateway_ncurses.cpp:714:44: warning: format ‘%d’ expects argument of type ‘int’, but argument 5 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=]
  714 |             mvwprintw(connPad, ctr++, 0, "%d %s\n", lCtr++, line.c_str());
      |                                           ~^        ~~~~~~
      |                                            |            |
      |                                            int          size_t {aka long unsigned int}
      |                                           %ld
```

Locally, I'm also seeing
```
/home/brendan/repos/RF24Gateway/examples/ncursesInt/RF24Gateway_ncursesInt.cpp:190:28: warning: ‘header.RF24NetworkHeader::type’ may be used uninitialized [-Wmaybe-uninitialized]
  190 |                 if (header.type == 1) // header.type is uninitialized
      |                     ~~~~~~~^~~~
/home/brendan/repos/RF24Gateway/examples/ncursesInt/RF24Gateway_ncursesInt.cpp:184:35: note: ‘header’ declared here
  184 |                 RF24NetworkHeader header;
      |                                   ^~~~~~
```
And rightfully so https://github.com/nRF24/RF24Gateway/blob/635e520e8c6746754d5799853aa551c30799a6dc/examples/ncursesInt/RF24Gateway_ncursesInt.cpp#L179-L208

Partly, I think this uninitialized `header.type` is a actually an oversight in `RF24NetworkHeader()` (empty c'tor). Although, the usage here makes a fresh instance of a header and emits a timestamp if the `header.type` is uninitialized. Logically speaking, this timestamp would be sent every time the header is constructed. So, I think the solution is just send the timestamp unconditionally (to master since `header.from_node` is supposedly also uninitialized).

--- a/RF24Gateway.cpp
+++ b/RF24Gateway.cpp
@@ -417,6 +417,7 @@ struct in_addr ESBGateway<mesh_t, network_t, radio_t>::getLocalIP()
     int family, s, n;
     char host[NI_MAXHOST];
     struct in_addr myNet;
+    memset(&myNet, 0, sizeof(myNet));
 
     getifaddrs(&ifap);
     for (ifa = ifap, n = 0; ifa != NULL; ifa = ifa->ifa_next, n++) {
@@ -682,14 +683,14 @@ template<class mesh_t, class network_t, class radio_t>
 void ESBGateway<mesh_t, network_t, radio_t>::printPayload(std::string buffer, std::string debugMsg)
 {
 }
-
+*/
 /***************************************************************************************
 
 template<class mesh_t, class network_t, class radio_t>
 void ESBGateway<mesh_t, network_t, radio_t>::printPayload(char* buffer, int nread, std::string debugMsg)
 {
 }
-
+*/
 /***************************************************************************************/
 
 template<class mesh_t, class network_t, class radio_t>

--- a/examples/ncurses/RF24Gateway_ncurses.cpp
+++ b/examples/ncurses/RF24Gateway_ncurses.cpp
@@ -372,9 +372,7 @@ void drawTopology()
 
             for (int j = 0; j < mesh.addrListTop; j++) {
                 if (mesh.addrList[j].address == i) {
-                    int y = 0;
-                    int x = 0;
-                    getyx(connPad, y, x);
+                    int x = getcurx(connPad);
                     if (x > connPadmaxX - 77) {
                         wprintw(connPad, "\n");
                     }
@@ -394,9 +392,7 @@ void drawTopology()
 
             for (int j = 0; j < mesh.addrListTop; j++) {
                 if (mesh.addrList[j].address == i) {
-                    int y = 0;
-                    int x = 0;
-                    getyx(connPad, y, x);
+                    int x = getcurx(connPad);
                     if (x > connPadmaxX - 77) {
                         wprintw(connPad, "\n");
                     }
@@ -416,9 +412,7 @@ void drawTopology()
 
             for (int j = 0; j < mesh.addrListTop; j++) {
                 if (mesh.addrList[j].address == i) {
-                    int y = 0;
-                    int x = 0;
-                    getyx(connPad, y, x);
+                    int x = getcurx(connPad);
                     if (x > connPadmaxX - 77) {
                         wprintw(connPad, "\n");
                     }
@@ -664,7 +658,7 @@ void drawRF24Pad()
     wprintw(rf24Pad, "TX Packets: %u\n", ok);
     wprintw(rf24Pad, "TX Drops: %u\n", fail);
 #endif
-    wprintw(rf24Pad, "RX Packets(user): %u\n", networkPacketsRX);
+    wprintw(rf24Pad, "RX Packets(user): %lu\n", networkPacketsRX);
 
     if (padSelection == 1)
     {
@@ -711,7 +705,7 @@ void drawConnPad()
             fnd = line.find_last_of(" ", fnd - 2);
             unsigned findEnd = line.find(" mark=");
             line = line.substr(fnd, findEnd - fnd);
-            mvwprintw(connPad, ctr++, 0, "%d %s\n", lCtr++, line.c_str());
+            mvwprintw(connPad, ctr++, 0, "%ld %s\n", lCtr++, line.c_str());
 
             if (ctr > maxX - 15)
             {

--- a/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
+++ b/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
@@ -400,9 +400,7 @@ void drawTopology()
 
             for (int j = 0; j < mesh.addrListTop; j++) {
                 if (mesh.addrList[j].address == i) {
-                    int y = 0;
-                    int x = 0;
-                    getyx(connPad, y, x);
+                    int x = getcurx(connPad);
                     if (x > connPadmaxX - 77) {
                         wprintw(connPad, "\n");
                     }
@@ -422,9 +420,7 @@ void drawTopology()
 
             for (int j = 0; j < mesh.addrListTop; j++) {
                 if (mesh.addrList[j].address == i) {
-                    int y = 0;
-                    int x = 0;
-                    getyx(connPad, y, x);
+                    int x = getcurx(connPad);
                     if (x > connPadmaxX - 77) {
                         wprintw(connPad, "\n");
                     }
@@ -444,9 +440,7 @@ void drawTopology()
 
             for (int j = 0; j < mesh.addrListTop; j++) {
                 if (mesh.addrList[j].address == i) {
-                    int y = 0;
-                    int x = 0;
-                    getyx(connPad, y, x);
+                    int x = getcurx(connPad);
                     if (x > connPadmaxX - 77) {
                         wprintw(connPad, "\n");
                     }
@@ -694,7 +688,7 @@ void drawRF24Pad()
     wprintw(rf24Pad, "TX Packets(sys): %u\n", ok);
     wprintw(rf24Pad, "TX Drops: %u\n", fail);
 #endif
-    wprintw(rf24Pad, "RX Packets(user): %u\n", networkPacketsRX);
+    wprintw(rf24Pad, "RX Packets(user): %lu\n", networkPacketsRX);
     if (gw.fifoCleared)
     {
         ++fifoClears;
@@ -747,7 +741,7 @@ void drawConnPad()
             fnd = line.find_last_of(" ", fnd - 2);
             unsigned findEnd = line.find(" mark=");
             line = line.substr(fnd, findEnd - fnd);
-            mvwprintw(connPad, ctr++, 0, "%d %s\n", lCtr++, line.c_str());
+            mvwprintw(connPad, ctr++, 0, "%ld %s\n", lCtr++, line.c_str());
 
             if (ctr > maxX - 15)
             {

--- a/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
+++ b/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
@@ -164,14 +164,12 @@ int main()
                 wclear(renewPad);
                 mvwprintw(renewPad, 0, 0, "*Renewing Address*");
                 prefresh(renewPad, 0, 0, 3, 26, 4, 55);
-                if (mesh.renewAddress() != MESH_DEFAULT_ADDRESS)
+                ok = mesh.renewAddress() != MESH_DEFAULT_ADDRESS;
+                if (ok)
                 {
                     ok = true;
                     wclear(renewPad);
                     prefresh(renewPad, 0, 0, 3, 26, 4, 55);
-                }
-                else {
-                    ok = false;
                 }
             }
             mesh_timer = millis();

--- a/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
+++ b/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
@@ -164,10 +164,14 @@ int main()
                 wclear(renewPad);
                 mvwprintw(renewPad, 0, 0, "*Renewing Address*");
                 prefresh(renewPad, 0, 0, 3, 26, 4, 55);
-                if ((ok = mesh.renewAddress()))
+                if (mesh.renewAddress() != MESH_DEFAULT_ADDRESS)
                 {
+                    ok = true;
                     wclear(renewPad);
                     prefresh(renewPad, 0, 0, 3, 26, 4, 55);
+                }
+                else {
+                    ok = false;
                 }
             }
             mesh_timer = millis();

--- a/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
+++ b/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
@@ -182,29 +182,26 @@ int main()
             {
                 ++networkPacketsRX;
                 RF24NetworkHeader header;
-
                 // un-needed variables
                 // size_t size = network.peek(header);
                 // uint8_t buf[size];
+                network.read(header, /*buf*/ 0, /*size*/ 0); // pop the RX payload from network.queue
 
-                if (header.type == 1) // header.type is uninitialized
+                // send a timestamp to master
+                struct timeStruct
                 {
-                    struct timeStruct
-                    {
-                        uint8_t hr;
-                        uint8_t min;
-                    } myTime;
+                    uint8_t hr;
+                    uint8_t min;
+                } myTime;
 
-                    time_t mTime;
-                    time(&mTime);
-                    struct tm* tm = localtime(&mTime);
+                time_t mTime;
+                time(&mTime);
+                struct tm* tm = localtime(&mTime);
 
-                    myTime.hr = tm->tm_hour;
-                    myTime.min = tm->tm_min;
-                    RF24NetworkHeader hdr(header.from_node, 1);
-                    network.write(hdr, &myTime, sizeof(myTime));
-                }
-                network.read(header, 0, 0);
+                myTime.hr = tm->tm_hour;
+                myTime.min = tm->tm_min;
+                RF24NetworkHeader hdr(0, 1);
+                network.write(hdr, &myTime, sizeof(myTime));
             }
         }
         else

--- a/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
+++ b/examples/ncursesInt/RF24Gateway_ncursesInt.cpp
@@ -200,7 +200,7 @@ int main()
 
                 myTime.hr = tm->tm_hour;
                 myTime.min = tm->tm_min;
-                RF24NetworkHeader hdr(0, 1);
+                RF24NetworkHeader hdr(header.from_node, 1);
                 network.write(hdr, &myTime, sizeof(myTime));
             }
         }

